common/features/transaction/sagas.ts

```diff
@@ -0,0 +1,894 @@
+  isValidHex,
+  isValidNonce,
+  gasPriceValidator,
+  gasLimitValidator
+} from 'libs/validators';
+import { transactionToRLP, signTransactionWithSignature } from 'utils/helpers';
+import { AppState } from 'features/reducers';
+import {
+  CONFIG_META,
+  ToggleAutoGasLimitAction,
+  getNodeLib,
+  isNetworkUnit,
+  getNetworkUnit,
+  getOffline,
+  getAutoGasLimitEnabled
+} from 'features/config';
+import { ENS, getResolvedAddress, resolveDomainRequested } from 'features/ens';
+import {
+  WALLET,
+  getWalletInst,
+  getToken,
+  getEtherBalance,
+  getCurrentBalance,
+  MergedToken,
+  getWalletType,
+  IWalletType
+} from 'features/wallet';
+import {
+  requestTransactionSignature,
+  PARITY_SIGNER as ParityKeys,
+  FinalizeSignatureAction
+} from 'features/paritySigner';
+import {
+  isSchedulingEnabled,
+  setSchedulingToggle,
+  setScheduleGasLimitField
+} from 'features/schedule';
+import { showNotification } from 'features/notifications';
+import { TRANSACTION_BROADCAST } from './broadcast';
+import {
+  TRANSACTION_FIELDS,
+  SetToFieldAction,
+  InputDataAction,
+  InputGasLimitAction,
+  InputGasPriceAction,
+  InputGasPriceIntentAction,
+  InputNonceAction,
+  SetDataFieldAction,
+  setToField,
+  setValueField,
+  inputGasPrice,
+  setDataField,
+  setGasLimitField,
+  setGasPriceField,
+  setNonceField,
+  inputNonce,
+  resetTransactionRequested,
+  resetTransactionSuccessful,
+  TSetValueField,
+  getTo,
+  getData,
+  getValue
+} from './fields';
+import {
+  TRANSACTION_META,
+  SetTokenToMetaAction,
+  SetTokenValueMetaAction,
+  SetUnitMetaAction,
+  setTokenTo,
+  setTokenValue,
+  setUnitMeta,
+  TSetTokenValue,
+  getDecimal,
+  getTokenValue,
+  getTokenTo,
+  isContractInteraction
+} from './meta';
+import {
+  TRANSACTION_NETWORK,
+  EstimateGasRequestedAction,
+  getFromSucceeded,
+  getFromFailed,
+  estimateGasFailed,
+  estimateGasSucceeded,
+  estimateGasRequested,
+  estimateGasTimedout,
+  getNonceSucceeded,
+  getNonceFailed
+} from './network';
+import {
+  TRANSACTION_SIGN,
+  SignWeb3TransactionSucceededAction,
+  SignLocalTransactionSucceededAction,
+  SignTransactionRequestedAction,
+  signLocalTransactionSucceeded,
+  signWeb3TransactionSucceeded
+} from './sign';
+import {
+  TRANSACTION,
+  SetCurrentToAction,
+  SetCurrentValueAction,
+  SwapEtherToTokenAction,
+  SwapTokenToTokenAction,
+  SwapTokenToEtherAction
+} from './types';
+import {
+  sendEverythingSucceeded,
+  sendEverythingFailed,
+  swapTokenToEther,
+  swapEtherToToken,
+  swapTokenToToken
+} from './actions';
+import {
+  isEtherTransaction,
+  getTransaction,
+  IGetTransaction,
+  getPreviousUnit,
+  getDecimalFromUnit,
+  getUnit,
+  getCurrentValue,
+  ICurrentValue,
+  getCurrentToAddressMessage,
+  serializedAndTransactionFieldsMatch
+} from './selectors';
+import {
+  IFullWalletAndTransaction,
+  signTransactionWrapper,
+  broadcastTransactionWrapper,
+  validateInput,
+  IInput,
+  rebaseUserInput
+} from './helpers';
```

Comment:
FYI: 177 lines of imports here.

Reply:
Restructuring to `import * as _ from _` will remove massive imports